src/shared_library.c

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "rcutils/error_handling.h"
+#include "rcutils/shared_library.h"
+#include "rcutils/strdup.h"
+
+rcutils_shared_library_t
+rcutils_get_zero_initialized_shared_library(void)
+{
+  rcutils_shared_library_t zero_initialized_shared_library;
+  zero_initialized_shared_library.library_path = NULL;
+  zero_initialized_shared_library.lib_pointer = NULL;
+  return zero_initialized_shared_library;
+}
+
+rcutils_ret_t
+rcutils_load_shared_library(
+  rcutils_shared_library_t * lib,
+  const char * library_path,
+  rcutils_allocator_t allocator)
+{
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(lib, RCUTILS_RET_INVALID_ARGUMENT);
+
+  lib->allocator = allocator;
+
+  lib->library_path = rcutils_strdup(library_path, lib->allocator);
```

Comment:
It needs to be either checked that `lib->library_path` is `NULL` before or it needs to be deallocated before.